packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -211,7 +272,12 @@ export class Interval implements ISerializableInterval {
 	 * @deprecated - This API was never intended to be public and will be marked internal in a future release.
 	 */
 	public propertyManager: PropertiesManager;
-	constructor(public start: number, public end: number, props?: PropertySet) {
+	constructor(
+		public start: number,
+		public end: number,
+		props?: PropertySet,
+		private readonly stickiness?: IntervalStickiness,
```

Comment:
In the sense that the stickiness is already implied by the sliding preference of the references? I had put stickiness on intervals with the assumption that the knowledge would be necessary for endpoint behavior -- stickiness at the interval level would determine whether the endpoints are exclusive or inclusive.

I was also thinking it would be necessary to track this directly for (de)serialization. I'll take a look at both of these cases.

Reply:
What is the distinction between `SequenceInterval` and `Interval`? I would expect that the regular `Interval` needs to track stickiness for the same reasons -- though, are regular `Interval`s never promoted to `SequenceInterval`s?

Reply:
regular `Interval` is an implementation of `IInterval` whose endpoints are plain old numbers. It's used for `SharedIntervalCollection`, a (deprecated) DDS which stores a collection of intervals like `[[5, 10], [0, 3], [9, 20]]` that has no underlying sequence. So it has nothing to do with local references, sliding, stickiness, merge-tree, etc.

Even though SharedIntervalCollection is deprecated, there's some value in making our interval type generic, since it can allow easier unit testing: for example consider Clarence's current work on interval indexes. We support an API for finding intervals that overlap a given range, and use a red-black tree to spatially accelerate that query. None of the interesting red-black tree logic is related to merge-tree or any of that complexity. So you can unit test that index with a lot less infrastructure by using `Interval` rather than `SequenceInterval`.

Reply:
Interesting -- my thinking was that `Interval` would still need stickiness to determine endpoint exclusivity, but in the use cases described it sounds like it would be fine to just always use inclusive endpoints, so it isn't necessary to track stickiness

Reply:
i don't think "inclusive" and "exclusive" even make sense here, since the numbers don't refer to anything. intervals could have negative or non-integer endpoints, even.